### DISCUSSION TITLE
Customize the snapshot of Core Plugins

### DISCUSSION THREAD

**Discussion by arturonavax** [LunarVim/LunarVim#4137] [Q&A] [answered]
Hi, is there any way to set the commit lock to a core plugin, I'm talking about the commits from [./snapshots/default.json](https://github.com/LunarVim/LunarVim/blob/master/snapshots/default.json), I use the stable version of LunarVim, but sometimes it seems that certain plugins are broken.

For example, at this very moment the [nvim-tree.lua](https://github.com/nvim-tree/nvim-tree.lua) plugin is configured in commit [`bb375fb`](https://github.com/nvim-tree/nvim-tree.lua/commit/bb375fb20327c49920c41d2b51c1ce2f4fe7deb3), and this version has a problem where the filesystem watcher is not working apparently, I manually go to `~/.local/share/lunarvim/site/pack/lazy/opt/nvim-tree.lua/` to perform a `git pull origin master` and it is fixed (commit [`270c955`](https://github.com/nvim-tree/nvim-tree.lua/commit/270c95556cad96d18ca547d86ae65927334b108b)).

Is there any way to do this from the user configuration in `~/.config/lvim/config.lua`? set the commit of a Core Plugin?

I know that you can set the commit of an extra plugin installed with [lazy.nvim](https://github.com/folke/lazy.nvim) with the "commit" attribute, but what would be the correct way to do that with a Core Plugin?

**Comment by LostInTheLogs** [comment, marked as answer]
https://www.lunarvim.org/docs/configuration/plugins/core-plugins#pinning

you can use `LVIM_DEV_MODE` env variable to unpin all plugins

**Reply by arturonavax** [reply]
Thank you very much for your help!

Is there any way to set specific commits to specific Core plugins? I'm not interested in unpining everyone to the latest version of the Core plugins.

**Reply by LostInTheLogs** [reply]
it's not possible

**Reply by arturonavax** [reply]
Thank you very much for helping me to solve this doubt

It would be great to be able to do this to solve problems like nvim-tree.lua, without having to have the latest (possibly unstable) version of the plugins.

**Reply by LostInTheLogs** [reply]
it will be possible in the future

**Reply by arturonavax** [reply]
For the moment I found a solution I think a little more feasible than what I had been doing.

The snapshot commit are in `~/.local/share/lunarvim/lvim/snapshots/default.json`, modifying this file is enough for `lvim +LvimSyncCorePlugins` to do the rest.
